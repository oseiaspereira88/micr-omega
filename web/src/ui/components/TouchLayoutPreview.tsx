import type { HTMLAttributes } from "react";
import styles from "./TouchLayoutPreview.module.css";

type TouchLayout = "left" | "right";

type TouchLayoutPreviewProps = {
  layout: TouchLayout;
  disabled?: boolean;
} & HTMLAttributes<HTMLDivElement>;

const TouchLayoutPreview = ({
  layout,
  disabled = false,
  className,
  "aria-hidden": ariaHidden,
  ...rest
}: TouchLayoutPreviewProps) => {
  const containerClassName = [
    styles.root,
    disabled ? styles.rootDisabled : "",
    className ?? "",
  ]
    .filter(Boolean)
    .join(" ");

  const activeLayout = layout === "left" ? "left" : "right";
  const options: TouchLayout[] = ["left", "right"];

  return (
    <div
      className={containerClassName}
      data-selected-layout={activeLayout}
      aria-hidden={ariaHidden ?? "true"}
      {...rest}
    >
      {options.map((option) => {
        const cardClassName = [
          styles.card,
          option === "left" ? styles.cardLeft : styles.cardRight,
          option === activeLayout ? styles.cardActive : styles.cardInactive,
        ]
          .filter(Boolean)
          .join(" ");

        const buttonsClassName = [
          styles.buttons,
          option === "left" ? styles.buttonsLeft : styles.buttonsRight,
        ]
          .filter(Boolean)
          .join(" ");

        return (
          <div
            key={option}
            className={cardClassName}
            data-layout={option}
            data-active={option === activeLayout ? "true" : "false"}
          >
            <div className={styles.cardContent}>
              <div className={styles.stick} />
              <div className={buttonsClassName}>
                <span className={styles.button} />
                <span className={`${styles.button} ${styles.buttonSecondary}`} />
                <span className={`${styles.button} ${styles.buttonTertiary}`} />
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default TouchLayoutPreview;
