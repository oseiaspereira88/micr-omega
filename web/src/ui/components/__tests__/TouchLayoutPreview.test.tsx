import { render, screen } from "@testing-library/react";

import TouchLayoutPreview from "../TouchLayoutPreview";

describe("TouchLayoutPreview", () => {
  it("gera snapshots distintos e nomes acessíveis para cada layout", () => {
    const { asFragment, rerender } = render(
      <TouchLayoutPreview layout="right" />
    );

    const rightPreview = screen.getByRole("img", {
      name: /prévia do layout com botões à direita/i,
    });
    expect(rightPreview).toHaveAttribute("data-layout", "right");
    expect(rightPreview).toHaveAccessibleName(
      /botões à direita/i
    );
    expect(asFragment()).toMatchSnapshot("touch-layout-preview-right");

    rerender(<TouchLayoutPreview layout="left" />);

    const leftPreview = screen.getByRole("img", {
      name: /prévia do layout com botões à esquerda/i,
    });
    expect(leftPreview).toHaveAttribute("data-layout", "left");
    expect(leftPreview).toHaveAccessibleName(/botões à esquerda/i);
    expect(asFragment()).toMatchSnapshot("touch-layout-preview-left");
  });
});
