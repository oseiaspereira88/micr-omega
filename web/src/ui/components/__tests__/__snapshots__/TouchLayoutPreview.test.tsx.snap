// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TouchLayoutPreview > gera snapshots distintos e nomes acessíveis para cada layout > touch-layout-preview-left 1`] = `
<DocumentFragment>
  <div
    aria-label="Prévia do layout com botões à esquerda"
    class="_preview_21c337"
    data-layout="left"
    data-touch-scale="1.000"
    data-touch-sensitivity="1.000"
    role="img"
    style="--touch-preview-scale: 1.000; --touch-preview-sensitivity: 1.000;"
  >
    <div
      class="_device_21c337"
    >
      <div
        class="_side_21c337 _leftSide_21c337"
      >
        <div
          class="_joystickRing_21c337"
        >
          <div
            class="_joystickHandle_21c337"
          />
        </div>
        <div
          class="_utilityRow_21c337"
        >
          <span
            aria-hidden="true"
            class="_utilityButton_21c337"
          />
          <span
            aria-hidden="true"
            class="_utilityBar_21c337"
          />
        </div>
      </div>
      <div
        class="_side_21c337 _rightSide_21c337"
      >
        <div
          class="_actionCluster_21c337"
        >
          <span
            aria-hidden="true"
            class="_actionButton_21c337"
          />
          <span
            aria-hidden="true"
            class="_actionButton_21c337"
          />
          <span
            aria-hidden="true"
            class="_actionButton_21c337"
          />
        </div>
        <span
          aria-hidden="true"
          class="_skillButton_21c337"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="_labels_21c337"
    >
      <span
        class="_label_21c337 _leftLabel_21c337"
      >
        Joystick
      </span>
      <span
        class="_label_21c337 _rightLabel_21c337"
      >
        Ações
      </span>
    </div>
    <div
      aria-hidden="true"
      class="_previewMeters_21c337"
    >
      <div
        class="_previewMeter_21c337"
      >
        <span
          class="_previewMeterLabel_21c337"
        >
          Tamanho
        </span>
        <div
          class="_previewMeterTrack_21c337"
        >
          <span
            class="_previewMeterFill_21c337"
            style="width: 100%;"
          />
        </div>
      </div>
      <div
        class="_previewMeter_21c337"
      >
        <span
          class="_previewMeterLabel_21c337"
        >
          Sensibilidade
        </span>
        <div
          class="_previewMeterTrack_21c337"
        >
          <span
            class="_previewMeterFill_21c337"
            data-variant="sensitivity"
            style="width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TouchLayoutPreview > gera snapshots distintos e nomes acessíveis para cada layout > touch-layout-preview-right 1`] = `
<DocumentFragment>
  <div
    aria-label="Prévia do layout com botões à direita"
    class="_preview_21c337"
    data-layout="right"
    data-touch-scale="1.000"
    data-touch-sensitivity="1.000"
    role="img"
    style="--touch-preview-scale: 1.000; --touch-preview-sensitivity: 1.000;"
  >
    <div
      class="_device_21c337"
    >
      <div
        class="_side_21c337 _leftSide_21c337"
      >
        <div
          class="_joystickRing_21c337"
        >
          <div
            class="_joystickHandle_21c337"
          />
        </div>
        <div
          class="_utilityRow_21c337"
        >
          <span
            aria-hidden="true"
            class="_utilityButton_21c337"
          />
          <span
            aria-hidden="true"
            class="_utilityBar_21c337"
          />
        </div>
      </div>
      <div
        class="_side_21c337 _rightSide_21c337"
      >
        <div
          class="_actionCluster_21c337"
        >
          <span
            aria-hidden="true"
            class="_actionButton_21c337"
          />
          <span
            aria-hidden="true"
            class="_actionButton_21c337"
          />
          <span
            aria-hidden="true"
            class="_actionButton_21c337"
          />
        </div>
        <span
          aria-hidden="true"
          class="_skillButton_21c337"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="_labels_21c337"
    >
      <span
        class="_label_21c337 _leftLabel_21c337"
      >
        Joystick
      </span>
      <span
        class="_label_21c337 _rightLabel_21c337"
      >
        Ações
      </span>
    </div>
    <div
      aria-hidden="true"
      class="_previewMeters_21c337"
    >
      <div
        class="_previewMeter_21c337"
      >
        <span
          class="_previewMeterLabel_21c337"
        >
          Tamanho
        </span>
        <div
          class="_previewMeterTrack_21c337"
        >
          <span
            class="_previewMeterFill_21c337"
            style="width: 100%;"
          />
        </div>
      </div>
      <div
        class="_previewMeter_21c337"
      >
        <span
          class="_previewMeterLabel_21c337"
        >
          Sensibilidade
        </span>
        <div
          class="_previewMeterTrack_21c337"
        >
          <span
            class="_previewMeterFill_21c337"
            data-variant="sensitivity"
            style="width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
